policy.go

```diff
@@ -62,67 +64,80 @@ func (policyJSON *Policy) UnmarshalJSON(policy []byte) error {
 	return err
 }
 
-// Parse decodifies input JSON info into Statement type
+// Parse decodes input JSON info into Statement type
 func (statementJSON *Statement) Parse(statement map[string]interface{}) {
 
 	// Definitions
 	var principal, notPrincipal, action, notAction, resource, notResource, condition []string
 	var err error
 
 	/* Iterate over map elements, each key element (statementKey) is the statement element
-	identifer and each value element (statementValue) the statement element value */
+	identifier and each value element (statementValue) the statement element value */
 	for statementKey, statementValue := range statement {
 		// Switch case over key type (identifying Statement elements)
 		switch statementKey {
 		case "StatementID":
 			// Type assertion to assign
 			statementJSON.StatementID = statementValue.(string)
+		case "Sid":
+			// Type assertion to assign
+			statementJSON.StatementID = statementValue.(string)
 		case "Effect":
 			//Type assertion to assign
 			statementJSON.Effect = statementValue.(string)
 		case "Principal":
 			// principal(statementValue) can be map[string][]string/string -> needs processing
 			// Initialize map
 			statementJSON.Principal = make(map[string][]string)
-			// procesing map
-			mapStatement := statementValue.(map[string]interface{})
-			// iterate over key principal (keyPrincipal) and value principal (valuePrincipal)
-			for keyPrincipal, valuePrincipal := range mapStatement {
-				// valuePrincipal can be string or []string
-				switch valuePrincipal := valuePrincipal.(type) {
-				case string:
-					// As map each element is identified with a key and has a value
-					principal = make([]string, 0)
-					statementJSON.Principal[keyPrincipal] = append(principal, valuePrincipal)
-				case []interface{}:
-					/* If value is an interface we know we have an []string -> knowing final type
-					we can use mapstructure (which uses reflect) to store as final type */
-					err = mapstructure.Decode(statementValue, &statementJSON.Principal)
-					if err != nil {
-						log.Error().Str("Error parsing policies", "Error using mapstructure parsing Policy statement principal element").Err(err).Msg("")
+			// processing map
+			switch statementValue.(type) {
```

Comment:
`Principal` and `NotPrincipal` can either be a map, or just `*` - https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_grammar.html